sdk/eventhub/azure-eventhubs/azure/eventhub/aio/_consumer_client_async.py

```diff
@@ -197,21 +200,22 @@ def from_connection_string(cls, conn_str: str,
             load_balancing_interval=load_balancing_interval,
             **kwargs
         )
+        return cls(**constructor_args)
 
     async def receive(
-            self, on_event,
+            self,
+            on_event: Callable[['PartitionContext', 'EventData'], None],
             *,
             partition_id: str = None,
             owner_level: int = None,
             prefetch: int = 300,
             track_last_enqueued_event_properties: bool = False,
-            starting_position=None,
-            starting_position_inclusive=False,
-            on_error=None,
-            on_partition_initialize=None,
-            on_partition_close=None
+            starting_position: Optional[Union[str, int, datetime.datetime, Dict[str, Any]]] = None,
+            starting_position_inclusive: Union[bool, Dict[str, bool]] = False,
```

Comment:
Optional?  (here and for http_proxy higher up in the file)

Reply:
Actually this one isn't optional because it can't be None (we have a default bool value).
However you're correct with http_proxy and a couple of other values I missed :)